sim/src/sensors/RotatingRaySensor.cpp

```diff
@@ -228,15 +262,55 @@ namespace mars {
 
       // Fills the pointcloud vector with (dist_m, x, y, z).
       // data[] contains all the measured distances.
-      for(unsigned int i=0; i<data.size(); i++) {
-        if (data[i] < config.maxDistance) {
-          // Calculates the ray/vector within the sensor frame.
-          utils::Vector local_ray = orientation_offset * directions[i] * data[i];
-          // Gathers pointcloud in the world frame to prevent/reduce movement distortion.
-          // This necessitates a back-transformation (world2node) in getPointcloud().
-          utils::Vector tmpvec = current_pose * local_ray;
-          pointcloud.push_back(tmpvec); // Scale normalized vector.
+      utils::Vector local_ray, tmpvec;
+      
+      // MultilevelLaserScan
+      velodyne_lidar::MultilevelLaserScan::SingleScan single_scan;
+      double dist_mm;
+      base::Angle h_angle, v_angle;
+      assert((int)data.size() == config.bands * config.lasers);
+      
+      // Fills the pointcloud vector with (dist_m, x, y, z).
+      // data[] contains all the measured distances according to the define directions.
+      int i = 0; // data_counter
+      for(int b=0; b<config.bands; ++b) { 
+        velodyne_lidar::MultilevelLaserScan::VerticalMultilevelScan vertical_scan;
+        vertical_scan.time = base::Time::now();
```

Comment:
Why is base::Time::now used here instead of the time getter we introducd to mars